### DISCUSSION TITLE
Including generateKeyPairSync of package crypto of Node

### DISCUSSION THREAD

**Discussion by dsportes** [quasarframework/quasar#10845] [Ideas / Proposals]
In node the **crypto** package includes the minimum set of functions for encryption. However the generateKeyPairSync is not avalaible in the crypto package embedded in Quasar. So it is possible in Quasar to encrypt / decrypt RSA but not to generate key pairs.
Please could you add this function ?

**Comment by ontwikkelfabriek** [comment]
@dsportes 

I am no cypto expert but here are some thoughts:

node -> runs server side ( here you generate your public/private keys with a crypto(js) package) 
quasar -> runs client side, you `cannot` run Nodejs `only` packages client side(Quasar), probably a very bad place to generate keys anyways.

Here's a read about Vue en crypto
https://codersdiaries.com/blog/vuejs-crypto

**Reply by dsportes** [reply]
Generating key pairs on client side is not a bad idea : the private key is so not avalaible in clear (not encrypted) on the server and users may trust that the server cannot store it somewhere. Anyway, it's a question of design.

Finally I found that crypto module is in fact the **crypto-browserify** module that _mimics_ few node equivalent functions with the same name and signature (not at all invoking directly node modules).
crypto-browserify is required by **node-libs-browser** required by webpack 4.44.1 required by @quasar/app !
So I'll contact authors of crypto-browserfiy.
Thanks for your time.

**Reply by ontwikkelfabriek** [reply]
@dsportes 

>  So I'll contact authors of crypto-browserfiy.

Good luck, looks like a **very** stale repo. 

I also found this code to generate public/private RSA key pairs in the browser:
https://github.com/diafygi/webcrypto-examples/#rsa-pss
